pygmt/src/solar.py

```diff
@@ -62,6 +64,34 @@ def solar(self, terminator="d", terminator_datetime=None, **kwargs):
     {panel}
     {perspective}
     {transparency}
+
+    Example
+    -------
+    >>> # import the Python module "datetime"
+    >>> import datetime
+    >>> import pygmt
+    >>> # create a datetime object at 8:52:18 on June 24, 1997 (time in UTC)
+    >>> date = datetime.datetime(
+    ...     year=1997, month=6, day=24, hour=8, minute=52, second=18
+    ... )
+    >>> # create a new plot with pygmt.Figure()
+    >>> fig = pygmt.Figure()
+    >>> # create a basemap of the Earth with the coast method
+    >>> fig.coast(
+    ...     land="lightgreen", water="lightblue", projection="W10c", region="d"
+    ... )
```

Comment:
Maybe we should only write `map`, because there is the separate `basemap` method?
```suggestion
    >>> # create a map of the Earth with the coast method
    >>> fig.coast(
    ...     land="lightgreen", water="lightblue", projection="W10c", region="d"
    ... )
```